testing/src/main/java/io/airlift/testing/Closeables.java

```diff
@@ -11,36 +8,44 @@ private Closeables()
     {
     }
 
-    public static void closeQuietly(Closeable... closeables)
+    public static void closeQuietly(AutoCloseable... closeables)
     {
         if (closeables == null) {
             return;
         }
-        for (Closeable closeable : closeables) {
+        for (AutoCloseable closeable : closeables) {
             try {
                 if (closeable != null) {
                     closeable.close();
                 }
             }
-            catch (IOException | RuntimeException ignored) {
+            catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                throw new RuntimeException(e);
+            }
+            catch (Exception ignored) {
             }
         }
     }
 
-    public static void closeAll(Closeable... closeables)
-            throws IOException
+    public static void closeAll(AutoCloseable... closeables)
+            throws Exception
     {
         if (closeables == null) {
             return;
         }
-        IOException rootCause = null;
-        for (Closeable closeable : closeables) {
+        Exception rootCause = null;
+        for (AutoCloseable closeable : closeables) {
             try {
                 if (closeable != null) {
                     closeable.close();
                 }
             }
-            catch (IOException e) {
+            catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                throw e;
```

Comment:
I think we should still continue to close everything, since these methods are used to prevent resource leaks. For `InterruptedException`, it probably makes sense to make this one the root exception. You could change the method to collect the exceptions into a list, then grab the first `InterruptedException` as the root one (if any).